tests/security_tests.rs

```diff
@@ -0,0 +1,305 @@
+    }
+
+    #[tokio::test]
+    async fn test_jwt_with_wrong_issuer() {
+        let shared_secret = "SECURE_SECRET_KEY_FOR_TESTING";
+        let jwt_codec = Arc::new(
+            JsonWebToken::<JwtClaims<Account<Role, Group>>>::new_with_options(JsonWebTokenOptions {
+                enc_key: EncodingKey::from_secret(shared_secret.as_bytes()),
+                dec_key: DecodingKey::from_secret(shared_secret.as_bytes()),
+                header: Some(Header::default()),
+                validation: Some(Validation::default()),
+            }),
+        );
+
+        let account = Account::new("user@example.com", &[Role::User], &[Group::new("test")]);
+        let registered_claims = RegisteredClaims::new(
+            "wrong-issuer", // Different issuer
+            (Utc::now() + TimeDelta::hours(1)).timestamp() as u64,
+        );
+        let claims = JwtClaims::new(account, registered_claims);
+
+        let jwt_bytes = jwt_codec.encode(&claims).unwrap();
+        let decoded_claims = jwt_codec.decode(&jwt_bytes).unwrap();
+
+        // Test issuer validation
+        assert!(!decoded_claims.has_issuer("expected-issuer"));
+        assert!(decoded_claims.has_issuer("wrong-issuer"));
+    }
+
+    #[tokio::test]
+    async fn test_jwt_with_different_secret() {
+        let secret1 = "SECRET_KEY_1";
+        let secret2 = "SECRET_KEY_2";
+        
+        let jwt_codec1 = Arc::new(
+            JsonWebToken::<JwtClaims<Account<Role, Group>>>::new_with_options(JsonWebTokenOptions {
+                enc_key: EncodingKey::from_secret(secret1.as_bytes()),
+                dec_key: DecodingKey::from_secret(secret1.as_bytes()),
+                header: Some(Header::default()),
+                validation: Some(Validation::default()),
+            }),
+        );
+
+        let jwt_codec2 = Arc::new(
+            JsonWebToken::<JwtClaims<Account<Role, Group>>>::new_with_options(JsonWebTokenOptions {
+                enc_key: EncodingKey::from_secret(secret2.as_bytes()),
+                dec_key: DecodingKey::from_secret(secret2.as_bytes()),
+                header: Some(Header::default()),
+                validation: Some(Validation::default()),
+            }),
+        );
+
+        let account = Account::new("user@example.com", &[Role::User], &[Group::new("test")]);
+        let registered_claims = RegisteredClaims::new(
+            "test-issuer",
+            (Utc::now() + TimeDelta::hours(1)).timestamp() as u64,
+        );
+        let claims = JwtClaims::new(account, registered_claims);
+
+        // Create JWT with first secret
+        let jwt_bytes = jwt_codec1.encode(&claims).unwrap();
+
+        // Try to decode with second secret - should fail
+        let result = jwt_codec2.decode(&jwt_bytes);
+        assert!(result.is_err(), "JWT signed with different secret should not be valid");
+    }
+}
+
+/// Tests for password hashing security and timing attack protection
+mod password_security_tests {
```

Comment:
Please check [`argon2`](https://github.com/RustCrypto/password-hashes/tree/master/argon2) repository for tests. Remove the tests in here if they are already tested in the external library.